internal/provider/network/routed_resource.go

```diff
@@ -0,0 +1,540 @@
+		networkMutexKV.KvLock(ctx, plan.EdgeGatewayID.ValueString())
+		defer networkMutexKV.KvUnlock(ctx, plan.EdgeGatewayID.ValueString())
+	}
+
+	ipPool := []staticIPPool{}
+	diags := plan.StaticIPPool.ElementsAs(ctx, &ipPool, true)
+	resp.Diagnostics.Append(diags...)
+
+	orgVDCNetworkConfig := &govcdtypes.OpenApiOrgVdcNetwork{
+		Name:        plan.Name.ValueString(),
+		Description: plan.Description.ValueString(),
+		OwnerRef:    &govcdtypes.OpenApiReference{ID: *parentEdgeGatewayOwnerID},
+
+		NetworkType: govcdtypes.OrgVdcNetworkTypeRouted,
+
+		// Connection is used for "routed" network
+		Connection: &govcdtypes.Connection{
+			RouterRef: govcdtypes.OpenApiReference{
+				ID: plan.EdgeGatewayID.ValueString(),
+			},
+			// API requires interface type in upper case, but we accept any case
+			ConnectionType: plan.InterfaceType.ValueString(),
+		},
+		Subnets: govcdtypes.OrgVdcNetworkSubnets{
+			Values: []govcdtypes.OrgVdcNetworkSubnetValues{
+				{
+					Gateway:      plan.Gateway.ValueString(),
+					PrefixLength: int(plan.PrefixLength.ValueInt64()),
+					DNSServer1:   plan.DNS1.ValueString(),
+					DNSServer2:   plan.DNS2.ValueString(),
+					DNSSuffix:    plan.DNSSuffix.ValueString(),
+					IPRanges: govcdtypes.OrgVdcNetworkSubnetIPRanges{
+						Values: processIPRanges(ipPool),
+					},
+				},
+			},
+		},
+	}
+
+	orgNetwork, err := org.CreateOpenApiOrgVdcNetwork(orgVDCNetworkConfig)
+	if err != nil {
+		resp.Diagnostics.AddError("Error creating routing network", err.Error())
+		return
+	}
+
+	plan = &networkRoutedResourceModel{
+		ID:            types.StringValue(orgNetwork.OpenApiOrgVdcNetwork.ID),
+		Name:          plan.Name,
+		Description:   plan.Description,
+		EdgeGatewayID: plan.EdgeGatewayID,
+		InterfaceType: plan.InterfaceType,
+		Gateway:       plan.Gateway,
+		PrefixLength:  plan.PrefixLength,
+		DNS1:          plan.DNS1,
+		DNS2:          plan.DNS2,
+		DNSSuffix:     plan.DNSSuffix,
+		StaticIPPool:  plan.StaticIPPool,
+	}
+
+	// Set state to fully populated data
+	resp.Diagnostics.Append(resp.State.Set(ctx, &plan)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+}
+
+// Read refreshes the Terraform state with the latest data.
+func (r *networkRoutedResource) Read(ctx context.Context, req resource.ReadRequest, resp *resource.ReadResponse) {
+	var state *networkRoutedResourceModel
+	// Get current state
+	resp.Diagnostics.Append(req.State.Get(ctx, &state)...)
+	if resp.Diagnostics.HasError() {
+		return
+	}
+
+	org, _, err := r.client.GetOrgAndVDC(r.client.GetOrg(), r.client.GetDefaultVDC())
+	if err != nil {
+		resp.Diagnostics.AddError("Error retrieving VDC", err.Error())
+		return
+	}
+
+	orgNetwork, err := org.GetOpenApiOrgVdcNetworkById(state.ID.ValueString())
+	if err != nil {
+		if govcd.ContainsNotFound(err) {
+			tflog.Debug(ctx, "Network not found, removing resource from state")
+			resp.State.RemoveResource(ctx)
+		}
+		resp.Diagnostics.AddError("Error retrieving routing network", err.Error())
+		return
+	}
+
```

Comment:
For preventing `nil` pointer check if `orgNetwork.OpenApiOrgVdcNetwork `, `orgNetwork.OpenApiOrgVdcNetwork.Connection`, `orgNetwork.OpenApiOrgVdcNetwork.Subnets` is not `nil`

And check if array `orgNetwork.OpenApiOrgVdcNetwork.Subnets.Values` array is at least 1